### DISCUSSION TITLE
converting jspsych data json output to csv

### DISCUSSION THREAD

**Discussion by HuangHam** [jspsych/jsPsych#2741] [Q&A] [answered]
Hi friends. I know that jspsych provides options to either save data in json format or csv format, however, for various reasons the software I'm using jspsych on can only support saving data as a json file. Does anyone have any preprocessing scripts (ideally in r), that can parse the json formatted data into csv? If you could share them with me it would be a huge help!

**Comment by jodeleeuw** [comment, marked as answer]
Hi @HuangHam 

I've had success using the {jsonlite} package to read json files into an R data frame. Then you can use R's built in `write.csv` or {readr}'s `write_csv` to export the file to csv format.

```r
library(jsonlite)
df <- fromJSON('path/to/file.json')
write.csv(df, 'path/to/file.csv')
```

**Reply by HuangHam** [reply]
cool! Thanks! fromJSON() looks like a very handy function. I was stuck with read_json() which doesn't gives me a clean format.
